src/content/insertable-streams/endtoend-encryption/js/worker.js

```diff
@@ -114,29 +114,37 @@ function decodeFunction(encodedFrame, controller) {
   controller.enqueue(encodedFrame);
 }
 
-onmessage = async (event) => {
-  const {operation} = event.data;
+function handleTransform(operation, readable, writable) {
   if (operation === 'encode') {
-    const {readable, writable} = event.data;
     const transformStream = new TransformStream({
       transform: encodeFunction,
     });
     readable
         .pipeThrough(transformStream)
         .pipeTo(writable);
   } else if (operation === 'decode') {
-    const {readable, writable} = event.data;
     const transformStream = new TransformStream({
       transform: decodeFunction,
     });
     readable
         .pipeThrough(transformStream)
         .pipeTo(writable);
-  } else if (operation === 'setCryptoKey') {
+  }
+}
+
+onmessage = (event) => {
```

Comment:
Can you add a comment above this to say // This is the handler that uses Transferable Streams?